DotNetNuke.Authentication.Github/Components/GithubClient.cs

```diff
@@ -55,19 +55,24 @@ public GithubClient(int portalId, AuthMode mode)
 
         #endregion
 
-        protected override TimeSpan GetExpiry(string responseText)
-        {
-            var jsonSerializer = new JavaScriptSerializer();
-            var tokenDictionary = jsonSerializer.DeserializeObject(responseText) as Dictionary<string, object>;
-
-            return new TimeSpan(0, 0, Convert.ToInt32(tokenDictionary["expires_in"]));
-        }
-
-        protected override string GetToken(string responseText)
-        {
-            var jsonSerializer = new JavaScriptSerializer();
-            var tokenDictionary = jsonSerializer.DeserializeObject(responseText) as Dictionary<string, object>;
-            return Convert.ToString(tokenDictionary["access_token"]);
-        }
-    }
+        protected override TimeSpan GetExpiry(string responseText)
+        {
+            int seconds;
+            if (int.TryParse(
+                HttpUtility.ParseQueryString(responseText)["expires_in"],
+                NumberStyles.Integer,
+                CultureInfo.InvariantCulture,
+                out seconds))
+            {
+                return TimeSpan.FromSeconds(seconds);
+            }
+
+            return TimeSpan.MinValue;

```

Comment:
You wrote "It doesn't look like an expiration is actually coming over, but I left it with a safe parsing attempt."  I agree, but this is causing an exception when I run this code: 

"The added or subtracted value results in an un-representable DateTime. Parameter name: value"

This is because Timespan.MinValue is a very large negative value! I think the correct return value should be TimeSpan.Zero.

Using this return value fixes the problem.